### PR TITLE
AttachBarcodes generic entrypoint -- fix bugs & add test

### PR DESCRIPTION
**Pull request by mlin**

### CODE REVIEW

File: src/sctools/platform.py [hunk 1]
```diff
@@ -935,15 +935,16 @@ def _make_tag_generators(
         barcode_args = {'fastq_files': r1}
 
         if i1:
-            barcode_args['embedded_barcodes'] = [cls.sample_barcode]
-            tag_generators.append(fastq.EmbeddedBarcodeGenerator(**barcode_args))
+            sample_barcode_args = dict(barcode_args)
```
Comment (ambrosejcarr) [inline src/sctools/platform.py]: Can you explain why this change was needed in the PR description? Otherwise I agree with the tests and their assertions.

File: src/sctools/test/test_entrypoints.py [hunk 2]
```diff
@@ -86,6 +86,61 @@ def test_Attach10XBarcodes_entrypoint_with_whitelist():
...
+
+    return_call = platform.BarcodePlatform.attach_barcodes(args)
+    assert return_call == 0
+    success = False
+    with pysam.AlignmentFile('test_tagged_bam.bam', 'rb', check_sq=False) as f:
+        for alignment in f:
+            if alignment.has_tag(consts.CELL_BARCODE_TAG_KEY):
+                success = True
+            # each alignment should now have a tag, and that tag should be a string
+            assert isinstance(alignment.get_tag(consts.RAW_CELL_BARCODE_TAG_KEY), str)
```
Comment (ambrosejcarr) [inline src/sctools/test/test_entrypoints.py]: Any interest in writing a small test that checks that the method is producing the _correct_ cell, molecule, and sample barcodes?

File: src/sctools/platform.py (hunk 1 above)
Comment (mlin) [reply src/sctools/platform.py]: OK

File: src/sctools/test/test_entrypoints.py (hunk 2 above)
Comment (mlin) [reply src/sctools/test/test_entrypoints.py]: Will follow up time permitting, thanks
